components/brave_wallet/browser/asset_ratio_response_parser.cc

```diff
@@ -424,4 +424,134 @@ mojom::BlockchainTokenPtr ParseTokenInfo(const std::string& json,
       *symbol, decimals, true, "", "", chain_id, coin);
 }
 
+bool ParseCoinMarkets(const std::string& json,
+                      std::vector<mojom::CoinMarketPtr>* values) {
+  DCHECK(values);
+  // {
+  //   "payload": [
+  //     {
+  //       "id": "bitcoin",
+  //       "symbol": "btc",
+  //       "name": "Bitcoin",
+  //       "image":
+  //       "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
```

Comment:
we need to ensure any external URLs in these responses also get proxied thru https://github.com/brave-intl/bat-go/pull/1234 to drop IP

Reply:
@fmarier suggested modifying the ratios server to rewrite these URLs to the proxy endpoint

Reply:
also curious what is the `1547033579` at the end for?

Reply:
> also curious what is the 1547033579 at the end for?

I'm not sure, but it looks like a timestamp.  If I were to guess, perhaps when coingecko updated the image last.  I tried a handful of URLs and they loaded the same without the query parameter, so we could try to strip it, but I'm not 100% sure on the effects of that.

Reply:
@nvonpentz it's more of a privacy concern. from my perspective it's not super important whether it gets done at the server or client as long as we ensure all these external URLs are getting proxied.

Reply:
Doing it on the client is faster to ship initially, but doing it server-side means you can push updates immediately instead of doing uplifts and waiting for the next release. So you could start with doing it client-side and then switch to server-side if it's something that ends up needing to be updated often.

Reply:
Ah, so I agree we should swap the asset URL on the server rather than the client.  I'm more curious if there's a faster way to do the server implementation, since we're writing the proxy endpoints by hand each time right now (like in https://github.com/brave-intl/bat-go/pull/1234). There's nothing generic that forwards requests to coingecko.  We don't have anything that handles `/coins/images/` right now.

It's not much to add that, but I'm wondering if we should set up something more generic (like we currently are doing with requests to Infura). Based on the discussion here https://github.com/brave/security/issues/784#issuecomment-1071180996, I was thinking the reason we wouldn't do something generic would be because of added security benefits of us parsing/validating coingecko response in the proxy before returning them to the client.  So I'm basically wondering if that's the motivation here as well.

Reply:
Implemented in https://github.com/brave-intl/bat-go/pull/1426. Would you please review @diracdeltas or @fmarier?

Reply:
Looks like we ended up going with https://github.com/brave/devops/issues/7563 to solve this problem and that's been merged. Going to go ahead and resolve this one.